### PR TITLE
Integration of Device Manager with ODIM's services

### PR DESCRIPTION
**Pull request by MaciejMis**
This pull request contains code related to integration between Device Manager and ODIM services moved into this project.
Most of the Device Manager code was rewritten. It connects with services via ODIM plugin interface using https.

Please note that most of the `files changed` comes from removing `/vendor` folder of Device Manager.
All code inside this PR was already reviewed on `integration` branch and the reviews can be found [here](https://github.com/opencomputeproject/HWMgmt-DeviceMgr-DeviceManager/pulls?q=is%3Apr+is%3Aclosed) but be welcomed to test and give us feedback!

Fixes:
#2
#19
#24
#29

### CODE REVIEW

File: svc-device-manager/data_collector.go
Comment (ehaligow) [inline svc-device-manager/data_collector.go:1]: There is one file [update_service.go] left in src folder, can we move it as well?

Comment (MaciejMis) [reply svc-device-manager/data_collector.go:1]: Fixed

File: README.md [hunk 1]
```diff
@@ -21,239 +21,131 @@ under the License.
...
+  - **Etcd**
+    - (project root)/build/etcd/etcd.yml
+
+## Install Device Manager using single command
+Use this command to install and run Device Manager together with ODIM services, Redis and Etcd as Docker containers.
+
+```shell
+$ make all
+```
+
```
Comment (jcleung5549) [inline README.md:61]: Add paragraph - "Once that make is complete, the Redfish interface will be available at 127.0.0.1:45000. The port number is specified by the odimra_config.json file, property APIGatewayConf.Port.  The Redfish interface is used to create a connection to Device Manager."

Comment (MaciejMis) [reply README.md:61]: Added

**Comment by MaciejMis** [comment]
This pull request was accepted on OpenRMC-DM meeting (25th April)

**Pull request merged**